### ISSUE TITLE
Could not find a declaration file for module 'webpack' / 'tapable'

### ISSUE THREAD

**Issue by IlCallo** [TypeStrong/fork-ts-checker-webpack-plugin#339]
## Current behavior
```
ERROR in /.../node_modules/fork-ts-checker-webpack-plugin/lib/hooks.d.ts(1,26):
TS7016: Could not find a declaration file for module 'webpack'. '/.../node_modules/webpack/lib/webpack.js' implicitly has an 'any' type.
  Try `npm install @types/webpack` if it exists or add a new declaration (.d.ts) file containing `declare module 'webpack';`
ERROR in /.../node_modules/fork-ts-checker-webpack-plugin/lib/hooks.d.ts(2,43):
TS7016: Could not find a declaration file for module 'tapable'. '/.../node_modules/tapable/lib/index.js' implicitly has an 'any' type.
  Try `npm install @types/tapable` if it exists or add a new declaration (.d.ts) file containing `declare module 'tapable';`
Version: typescript 3.6.3
Time: 5433ms
```

## Expected behavior
No errors by default, or a `peerDependencies` warning fired off, or a disclaimer in setup section of the README.

## Steps to reproduce the issue
Not really sure, I just updated `fork-ts-checker-webpack-plugin` to `1.5.0` from `0.5.2` and the error popped up.
The error doesn't show up in `1.3.1` because of the typings dependencies being added, then removed again in `1.3.2`.
The problematic declaration seems to be into `lib/hooks.d.ts`, where `webpack` and `tapable` are types are imported.

```ts
import * as webpack from 'webpack';
import { AsyncSeriesHook, SyncHook } from 'tapable';
export declare type ForkTsCheckerHooks = 'serviceBeforeStart' | 'cancel' | 'serviceStartError' | 'waiting' | 'serviceStart' | 'receive' | 'serviceOutOfMemory' | 'emit' | 'done';
declare type ForkTsCheckerLegacyHookMap = Record<ForkTsCheckerHooks, string>;
export declare const legacyHookMap: ForkTsCheckerLegacyHookMap;
export declare function getForkTsCheckerWebpackPluginHooks(compiler: webpack.Compiler): Record<ForkTsCheckerHooks, SyncHook<any, any, any> | AsyncSeriesHook<any, any, any>>;
export {};
```

Manually adding the dependencies fix the errors.

Related:
- https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/192
- https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/pull/281

## Environment
 - **fork-ts-checker-webpack-plugin**: 1.5.0
 - **typescript**: 3.6.3
 - **tslint**: N/A
 - **eslint**: 5.10.0
 - **webpack**: 4.41.0
 - **os**: Ubuntu 18.04.3 LTS

**Comment by johnnyreilly** [comment]
Oh that's a shame. If you'd like to reattempt https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/pull/281 (I'm not sure how this regressed) I'll certainly take a look at it 🥰

**Comment by phryneas** [comment]
In #287 I had moved everything depending on webpack/tapable out to hooks.d.ts in the hopes that only index.d.ts would be included automatically and the rest just where used.

We should definitely not add @types/webpack as a dependency, as as I remember that would re-introduce @types/node as a dependency.

Maybe just move the declaration 
```typescript
export declare type ForkTsCheckerHooks = 'serviceBeforeStart' | 'cancel' | 'serviceStartError' | 'waiting' | 'serviceStart' | 'receive' | 'serviceOutOfMemory' | 'emit' | 'done';
```
to the index.d.ts, reference it from hooks.ts and make sure hooks.d.ts is removed on build, as it won't contain any more external references?

**Comment by IlCallo** [comment]
> If you'd like to reattempt #281

Not sure about how your typings generation works. If @phryneas already have an idea on it, I'll be happy to check if it works.

I think https://github.com/microsoft/TypeScript/issues/18588 is somewhat related and there could be some workarounds. Still, no official solution

**Comment by akleiber** [comment]
hi @phryneas do you have a PR for this that I may not find? Because you mentioned here https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/pull/342#issuecomment-545757062 that you are working on it - if not is your suggestion solution the way to go?
-> https://github.com/TypeStrong/fork-ts-checker-webpack-plugin/issues/339#issuecomment-539119519

**Comment by phryneas** [comment]
@akleiber  Argh, I had experimented using [API-extractor](https://api-extractor.com/pages/setup/configure_rollup/) for that, hit a roadblock, got distracted and forgot about that. Sorry!
Yes, comment 339 would be the easy way to go about this. Would you want to do a PR on this? I'm a bit swamped right now.

**Comment by piotr-oles** [comment]
Fix released in the 5.0.0-alpha.7 version :)
